### PR TITLE
drop deprecated result object 1/n

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

remove usage of returning ResultObject from the train loop

### Needed pre-steps (particular PRs):
- [x] drop unused API with results in tests
- [ ] drop usage as return output in tests
- [x] drop TrainResult >> #5323
- [x] drop EvalResult >> #5633

## Before submitting
- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? 
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:  

 - [ ] Is this pull request ready for review? (if not, please submit in draft mode)
 - [ ] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified; _Bugfixes should be including in bug-fix release milestones (m.f.X) and features should be included in (m.X.b) releases._
 

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by Borda** [comment]
the point with failing tests is that we use Train/EvalResult in about 60 places in tests...

File: pytorch_lightning/trainer/connectors/logger_connector/logger_connector.py [hunk 1]
```diff
@@ -392,31 +392,14 @@ def log_train_epoch_end_metrics(
...
-            self.training_epoch_end(model, epoch_output, num_optimizers)
-
-            # log/aggregate metrics automatically
-            epoch_log_metrics, epoch_progress_bar_metrics = self.__auto_reduce_results_on_epoch_end(epoch_output)
+        # lightning module hook
+        self.training_epoch_end(model, epoch_output, num_optimizers)
 
-        # TODO: deprecate 1.0
-        else:
-            out = self.__run_legacy_training_epoch_end(
```
Comment (carmocca) [inline pytorch_lightning/trainer/connectors/logger_connector/logger_connector.py:416]: Remove this function too? is it used anywhere else?

Comment (Borda) [reply pytorch_lightning/trainer/connectors/logger_connector/logger_connector.py:416]: probably yes, just will do in smaller steps...

File: pytorch_lightning/trainer/connectors/logger_connector/logger_connector.py [hunk 2]
```diff
@@ -395,31 +395,14 @@ def log_train_epoch_end_metrics(
...
-            sample_obj = opt_idx_outputs[0][0] if isinstance(opt_idx_outputs[0], list) else opt_idx_outputs[0]
-            is_result_obj = len(epoch_output) > 0 and isinstance(sample_obj, Result)
-            is_1_0_result = is_result_obj and 'extra' in sample_obj
-        except IndexError:
-            is_result_obj = False
-            is_1_0_result = False
-
         # ------------------
         # NEW 1.0.0 PATH
         # ------------------
```
Comment (carmocca) [inline pytorch_lightning/trainer/connectors/logger_connector/logger_connector.py:400]: Remove this?

```suggestion
```

**Comment by tchaton** [comment]
Hey @Borda, mind resolving the failing tests ?

**Comment by Borda** [comment]
@carmocca @tchaton mind check why the tests are failing, what else is missing?

**Comment by carmocca** [comment]
@Borda What tests are you referring to in particular?

**Review by SeanNaren — APPROVED** [review]

**Review by awaelchli — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
LGTM !
